inference-engine/src/low_precision_transformations/src/network_helper.cpp

```diff
@@ -189,52 +189,66 @@ std::shared_ptr<Node> NetworkHelper::swapMultiplyAndAdd(std::shared_ptr<opset1::
+
+    for (size_t i = 0; i < newShiftsValues.size(); ++i) {
+        const auto shiftsValue = shiftsValues[shiftsAreBroadcasted ? 0 : i];
+        const auto scalesValue = scalesValues[scalesAreBroadcasted ? 0 : i];
+
+        const auto absScalesValue = std::abs(static_cast<double>(scalesValue));
+        if ((scalesValue == 0.) || ((absScalesValue > 0.) && (absScalesValue < 1.e-32))) {
+            if (newScalesValues.empty()) {
+                newScalesValues = scalesAreBroadcasted ? std::vector<float>(shiftsValues.size(), scalesValues[0]) : scalesValues;
             }
+            newShiftsValues[i] = shiftsValues[shiftsAreBroadcasted ? 0 : i];
+            newScalesValues[i] = 1.f;
+        } else {
+            newShiftsValues[i] = shiftsValue / scalesValue;
         }
-
-        bDivA = std::make_shared<opset1::Constant>(
-                b->get_output_element_type(0),
-                aBroadcasted ? b->get_output_shape(0) : a->get_output_shape(0),
-                bDivAValues);
-    } else {
-        bDivA = fold<opset1::Divide>(b, a);
     }
 
-    std::vector<std::shared_ptr<Node>> inputs{ {}, {} };
+    newShifts = std::make_shared<opset1::Constant>(
+            shifts->get_output_element_type(0),
+            scalesAreBroadcasted ? shifts->get_output_shape(0) : scales->get_output_shape(0),
+            newShiftsValues);
 
-    inputs[0] = x;
-    inputs[1] = bDivA;
+    std::vector<std::shared_ptr<Node>> newAddInputs{ {}, {} };
+
+    newAddInputs[0] = x;
+    newAddInputs[1] = newShifts;
 
     std::shared_ptr<opset1::Add> newAdd = std::make_shared<op::TypeRelaxed<opset1::Add>>(
         std::vector<element::Type>{element::f32, element::f32}, std::vector<element::Type>{ element::f32 },
-        ngraph::op::TemporaryReplaceOutputType(inputs[0], element::f32).get(),
-        ngraph::op::TemporaryReplaceOutputType(inputs[1], element::f32).get());
+        ngraph::op::TemporaryReplaceOutputType(newAddInputs[0], element::f32).get(),
+        ngraph::op::TemporaryReplaceOutputType(newAddInputs[1], element::f32).get());
     copyInfo(addAfterMultiply, newAdd);
 
     NetworkHelper::setOutDataPrecision(newAdd, addAfterMultiply->get_output_element_type(0));
 
-    auto newMultiply = std::make_shared<DequantizationMultiply>(newAdd, a);
+    if (!newScalesValues.empty() &&
+        std::all_of(
+            newScalesValues.begin(),
+            newScalesValues.end(),
+            [&newScalesValues](const float value) { return value == newScalesValues[0]; })) {
+        newScalesValues = { newScalesValues[0] };
+    }
+
+    auto newMultiply = std::make_shared<DequantizationMultiply>(
+        newAdd,
+        newScalesValues.empty() ? scales : std::make_shared<opset1::Constant>(
+            scales->get_output_element_type(0),
+            (scales->get_output_shape(0).empty() || scales->get_output_shape(0).size() == 1ul) ? Shape{} : scales->get_output_shape(0),
```

Comment:
Please, use `ngraph::shape_size(scales->get_output_shape(0) == 1)` instead of `scales->get_output_shape(0).empty() || scales->get_output_shape(0).size() == 1ul`